pkg/operator/vsphereproblemdetector/monitoring.go

```diff
@@ -0,0 +1,80 @@
+package vsphereproblemdetector
+
+import (
+	"context"
+	"time"
+
+	operatorapi "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-storage-operator/pkg/csoclients"
+	"github.com/openshift/cluster-storage-operator/pkg/generated"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+)
+
+type monitoringController struct {
+	operatorClient v1helpers.OperatorClient
+	kubeClient     kubernetes.Interface
+	dynamicClient  dynamic.Interface
+	eventRecorder  events.Recorder
+}
+
+const (
+	monitoringControllerName = "VSphereProblemDetectorMonitoringController"
+)
+
+func newMonitoringController(
+	clients *csoclients.Clients,
+	eventRecorder events.Recorder,
+	resyncInterval time.Duration) factory.Controller {
+
+	c := &monitoringController{
+		operatorClient: clients.OperatorClient,
+		kubeClient:     clients.KubeClient,
+		dynamicClient:  clients.DynamicClient,
+		eventRecorder:  eventRecorder,
+	}
+	return factory.New().
+		WithInformers(c.operatorClient.Informer()).
```

Comment:
Should it have watch informer of `ServiceMonitors` too?

Reply:
We could but it will require importing all of - https://github.com/prometheus-operator/prometheus-operator/tree/master/pkg/apis/monitoring but I was being thwarted by dependency conflicts, so I have avoided doing this. 

We are doing what we did for syncing credentials (before we moved them to CVO) and that is the reason we are rescyncing every 1 minute because we are not watching `ServiceMonitor` objects.

Reply:
I did ended up importing those informer and APIs and hence marking this as resolved.